eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -1543,3 +1562,151 @@ def test_info_view_success(self) -> None:
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("earned_grade", response_content)
+        if extra_params["detailed"]:
+            self.assertIn("section_breakdown", response_content)
+        else:
+            self.assertNotIn("section_breakdown", response_content)
+
+        if extra_params["grading_policy"]:
+            self.assertIn("grading_policy", response_content)
+            self.assertIn("grader", response_content["grading_policy"])
+            self.assertIn("grade_cutoffs", response_content["grading_policy"])
+        else:
+            self.assertNotIn("grading_policy", response_content)
+
+    def test_get_grade_for_user_not_found(self):
+        """
+        Test get grade info for a user and course from another site.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response contains an error message.
+        """
+        params = self.params.copy()
+        params.update({
+            "detailed": True,
+            "grading_policy": True,
+        })
+
+        response = self.get_grade_info(self.tenant_y, params)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn("detail", response_data)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'username': '{params['username']}'}} on site {self.tenant_y['domain']}.",
+        )
+
+    def test_get_grade_for_enrollment_not_found(self):
+        """
+        Test grade info for a enrollment not found.
+
+        Open edX definitions tested:
+        - `get_enrollment`
+
+        Expected result:
+        - The status code is 404.
+        - The response contains an error message.
+        """
+        response = self.get_grade_info(self.tenant_x, self.params)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data, [
+                f"No enrollment found for user:`{self.user['username']}`"
+            ],
+        )
+
+    def test_get_grade_for_course_not_found(self):
+        """
+        Test grade info for a course not found.
+
+        Open edX definitions tested:
+        - `get_enrollment`
+
+        Expected result:
+        - The status code is 404.
+        - The response contains an error message.
+        """
+        response = self.get_grade_info(
+            self.tenant_x,
+            {
+                **self.params,
+                "course_id": "course-not-exist"
+            }
+        )
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data, [
+                "No course found for course_id `course-not-exist`"
+            ],
+        )
```

Comment:
What would happen if the user does not exist?

Reply:
@mariajgrimaldi It seems to me that this is tested in the test `test_get_grade_for_user_not_found`, the expected result is the same...